skfda/exploratory/visualization/representation.py

```diff
@@ -333,143 +355,200 @@ def plot(
 
             X, Y = np.meshgrid(x, y, indexing='ij')
 
+            ind = 0
             for k in range(self.fdata.dim_codomain):
+                self.artists[k] = np.zeros(self.n_samples(), dtype=Artist)
                 for h in range(self.fdata.n_samples):
 
                     set_color_dict(sample_colors, h, color_dict)
 
-                    axes[k].plot_surface(
+                    self.artists[k][h] = self.axes[k].plot_surface(
                         X,
                         Y,
                         Z[h, ..., k],
                         **color_dict,
                         **kwargs,
                     )
+                    ind += 1
 
-        _set_labels(self.fdata, fig, axes, patches)
+        _set_labels(self.fdata, self.fig, self.axes, patches)
+        self.fig.suptitle("GraphPlot")
```

Comment:
Why? We should use the name of the dataset by default. And I think that is what `_set_labels` does.

Reply:
true